waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -826,15 +826,20 @@
 
 (defn attach-envoy-sidecar
   "Attaches envoy sidecar to replicaset"
-  [replicaset reverse-proxy service-description base-env service-port port0]
+  [replicaset reverse-proxy
+   {:strs [backend-proto health-check-port-index] :as service-description}
+   base-env service-port port0]
   (update-in replicaset
     [:spec :template :spec :containers]
     conj
     (let [{:keys [cmd resources image]} reverse-proxy
           user-env (:env service-description)
           env-map (-> user-env
-                  (merge base-env)
-                  (assoc "PORT0" (str port0) "SERVICE_PORT" (str service-port)))
+                      (merge base-env)
+                      (assoc "HEALTH_CHECK_PORT_INDEX" (str health-check-port-index)
+                             "PORT0" (str port0)
+                             "SERVICE_PORT" (str service-port)
+                             "SERVICE_PROTOCOL" backend-proto))
```

Comment:
The `backend-proto` may need to be capitalized here.

Reply:
No longer needed since we're consistently using lower-case again.